armi/physics/neutronics/fissionProductModel/fissionProductModel.py

```diff
@@ -205,60 +192,146 @@ def _getAllFissionProductNames(self):
+            numberDensities = {}
+            for nb in lfp.keys():
+                if not lumpedFissionProduct.isGas(nb):
+                    continue
+                yld = lfp[nb]
+                ndens = b.getNumberDensity(lfp.name)
+                numberDensities[nb.name] = ndens * yld
+            return numberDensities
+
+        def _removeFissionGasesExplicitFissionProductModeling(
+            core, gasRemovalFractions
+        ):
+            """
+            This is called when `explicitFissionProducts` are selected as the fission product model.
+
+            Notes
+            -----
+            The input provided has the amount of gas to be removed by each block in the core. The
+            gas that remains in a block is first calculated and then the number density of the
+            gaseous nuclides is multiplied by the fraction that remains and the update number
+            density vector for the block is updated.
+            """
+            for b in core.getBlocks():
+                if b not in gasRemovalFractions:
+                    continue
+                removedFraction = gasRemovalFractions[b]
+                remainingFraction = 1.0 - removedFraction
+                updatedNumberDensities = {}
+                for nuc, val in b.getNumberDensities().items():
+                    nb = nuclideBases.byName[nuc]
+                    updatedNumberDensities[nuc] = (
+                        remainingFraction * val
+                        if lumpedFissionProduct.isGas(nb)
+                        else val
+                    )
+                b.updateNumberDensities(updatedNumberDensities)
+
+        def _removeFissionGasesLumpedFissionProductModeling(core, gasRemovalFractions):
+            weightedGasRemoved = collections.defaultdict(float)
+            totalWeight = collections.defaultdict(float)
+            for b in core.getBlocks():
+                lfpCollection = b.getLumpedFissionProductCollection()
+                if lfpCollection is None or b not in gasRemovalFractions:
+                    continue
+
+                numberDensities = b.getNumberDensities()
+                for lfp in lfpCollection.values():
+                    ndens = _getGaseousFissionProductNumberDensities(b, lfp)
+                    removedFraction = gasRemovalFractions[b]
+                    remainingFraction = 1.0 - removedFraction
+
+                    # If the lumped fission products are global then we are going
+                    # release the average across all the blocks in the core and
+                    # this data is collected iteratively.
+                    if self._useGlobalLFPs:
+                        weight = b.getVolume() * (b.p.flux or 1.0)
+                        weightedGasRemoved[lfp] += (
+                            sum(ndens.values()) * removedFraction * weight
+                        )
+                        totalWeight[lfp] += weight
```

Comment:
Sure thing. I actually think that this caveat should be added to the `FissionProductModel` class docstring since LFPs that don't track `Xe` and `Sm` directly/explicitly would be wrong anyways.